libraries/hotstuff/chain_pacemaker.cpp

```diff
@@ -108,6 +108,7 @@ namespace eosio { namespace hotstuff {
       _accepted_block_connection = chain->accepted_block.connect( [this]( const block_state_ptr& blk ) {
          on_accepted_block( blk );
       } );
+      _head_block_state = chain->head_block_state();
```

Comment:
I think it needs a `std::scoped_lock g( _chain_state_mutex );` here as well

Reply:
In the constructor, so not needed.